freshmaker/handlers/botas/botas_shipped_advisory.py

```diff
@@ -115,6 +160,27 @@ def handle(self, event):
         bundles_nvrs = self._filter_bundles_by_pinned_related_images(
             auto_rebuild_nvrs)
 
+        # Now we have the bundle NVRs needs to be rebuilt, for each bundle NVR,
+        # construct the data of related images pullspec overrides
+        pullspec_overrides_by_nvr = {}
+        for bundle_nvr in bundles_nvrs:
+            images = [i for i in bundle_images if i['brew']['build'] == bundle_nvr]
+            # images are for different arches, but they have same value of
+            # manifest_list_digest, just get the value from the first image
+            bundle_digest = images[0]['manifest_list_digest']
+
+            # check which original images connect to this bundle image
+            for image_nvr, digests in bundle_digests_by_related_nvr.items():
+                if bundle_digest not in digests:
+                    continue
+
+                # this original image NVR connects to bundle_nvr
+                original_digest = original_digests_by_nvr[image_nvr]
+                rebuilt_digest = rebuilt_digests_by_nvr[nvrs_mapping[image_nvr]]
+
+                pullspec_overrides_by_nvr.setdefault(bundle_nvr, {})
+                pullspec_overrides_by_nvr[bundle_nvr][original_digest] = rebuilt_digest
```

Comment:
This is because we're adding multiple `original_digest: rebuilt_digest` to a single bundle image within the loop of 
```
for image_nvr, digests in bundle_digests_by_related_nvr.items():
```
multiple rebuilt images in an advisory can connect to a same bundle image, this is a common case.

Reply:
Oh ok, now I got it. Thx